contracts/partial/parser/vendor/harbinger/methods_callback.ligo

```diff
@@ -79,21 +88,32 @@ function receivePrice(
     const usd : bool = (assetName = xtz_usd_price_name); // if price is XTZ/USD
     var priceF : precisionValue := 0n;
     if (usd) then {
-        var data := unpackTmpOrDefault(s.tmp);
-        require(data.updating, Errors.xtz_not_updating);
-        priceF := s.oraclePrecision * precision / oraclePrice;
-        data := record[
-          price = oraclePrice;
-          level = Tezos.level;
-          updating = False;
+        var data := unpackTmp(s.tmp);
+        case data.state of [
+          WaitingUsdPrice -> skip
+          | _ -> failwith(Errors.invalidState)
         ];
+        priceF := s.oraclePrecision * precision / oraclePrice;
+        data.price := oraclePrice;
+        data.level := Tezos.level;
+        data.state := WaitingAssetPrice(0n);
```

Comment:
I don't understand how this works.
Isn't the number in payload supposed to be total count of callbacks we expect to be invoked?

Reply:
Nope, num in payload describes current state and index. Field 'size' is responsible for  storing count of callbacks